### PR TITLE
Replaced copy_with_new_args() to clone_with_new_inputs()

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

**Comment by gladilov-gleb** [comment]
You removed binary files from `docs/img/`. Is it expected?

File: inference-engine/src/inference_engine/generic_ie.cpp [hunk 1]
```diff
@@ -57,19 +57,14 @@ std::vector<InferenceEngine::IShapeInferExtensionPtr> ngraph::op::GenericIE::get
     return extensions;
 }
 
-ngraph::op::GenericIE::GenericIE(const ngraph::NodeVector& inputs,
-                                 const std::map<std::string, InferenceEngine::Parameter>& params,
-                                 const std::string type, const std::vector<PortIE>& outputs)
-    : GenericIE(as_output_vector(inputs), params, type, outputs) {}
-
```
Comment (gladilov-gleb) [inline inference-engine/src/inference_engine/generic_ie.cpp:64]: Why you removed this ctor?

File: ngraph/src/ngraph/node.cpp [hunk 2]
```diff
@@ -124,29 +124,14 @@ std::shared_ptr<Node>
...
-            clone->set_argument(i, inputs.at(i));
-        }
-    }
-    return clone;
+    NODE_VALIDATION_CHECK(this,
+                          false,
+                          "Internal error: clone_with_new_inputs not implemented for node ",
+                          get_friendly_name(),
+                          " with type ",
+                          description());
```
Comment (gladilov-gleb) [inline ngraph/src/ngraph/node.cpp:134]: Why you removed `clone_with_new_inputs` implementation for `ngraph::Node`?

File: ngraph/src/ngraph/node.hpp [hunk 3]
```diff
@@ -400,14 +400,7 @@ namespace ngraph
...
         Output<Node> get_input_source_output(size_t i) const;
 
-    protected:
-        // Will be replaced with clone_with_new_inputs
-        virtual std::shared_ptr<Node> copy_with_new_args(const NodeVector& new_args) const
-            NGRAPH_DEPRECATED("use copy_with_new_inputs instead");
-
     public:
-        // TODO: When all copy_with_new_args have been replaced with copy_with_new_inputs, make
-        // this pure and remove copy_with_new_args
```
Comment (gladilov-gleb) [inline ngraph/src/ngraph/node.hpp:410]: You removed `TODO`, but didn't make method _pure_. Why?

File: ngraph/test/op.cpp [hunk 4]
```diff
@@ -114,21 +114,3 @@ TEST(op, variant)
...
-    class MockOp: public op::Op
-    {
-        MOCK_CONST_METHOD1(copy_with_new_args, std::shared_ptr<Node>(const NodeVector& new_args));
-        MOCK_CONST_METHOD1(get_users, NodeVector (bool check_is_used)); // This can't be mocked as
-                                                                        // it's non-virtual
-    };
-
-    ::testing::NiceMock<MockOp> mock_op;
-}
-*/
```
Comment (gladilov-gleb) [inline ngraph/test/op.cpp:134]: Why you removed this test?

File: ngraph/src/ngraph/node.cpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/src/ngraph/node.cpp:134]: `clone_with_new_inputs` is deprecated method which cannot work with operations which have several outputs.

File: ngraph/src/ngraph/node.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/src/ngraph/node.hpp:410]: I added exception in this method. Good point. Will change to pure.

File: ngraph/test/op.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/op.cpp:134]: This test were commented. I just removed redundant commented code.

**Comment by ilyachur** [comment]
> * You removed binary files from `docs/img/`. Is it expected?

No it is a merge conflict. Will revert.

File: inference-engine/src/inference_engine/generic_ie.cpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/inference_engine/generic_ie.cpp:64]: We have the same constructor which use vector of outputs.

File: ngraph/test/op.cpp (hunk 4 above)
Comment (gladilov-gleb) [reply ngraph/test/op.cpp:134]: Is there ticket that will remind to enable such test in future?

Comment (ilyachur) [reply ngraph/test/op.cpp:134]: No this is nGraph mechanism which is not used in the OpenVINO.

**Comment by ilyachur** [comment]
> * PR title says you removed method, but in fact, you replaced it with another.

The name of PR was changed

File: ngraph/test/op.cpp (hunk 4 above)
Comment (gladilov-gleb) [reply ngraph/test/op.cpp:134]: But if there is still such mechanism should it be covered by tests even if now it is unused in OpenVINO? If you consider mechanism useless should it be removed with rest as well (or you're doing it right now removing `copy_with_new_args`)?

File: ngraph/src/ngraph/node.cpp (hunk 2 above)
Comment (gladilov-gleb) [reply ngraph/src/ngraph/node.cpp]: If `clone_with_new_inputs` is deprecated why you replace `copy_with_new_args` with it?

Comment (ilyachur) [reply ngraph/src/ngraph/node.cpp]: Sorry my mistake. `copy_with_new_args` is deprecated, `clone_with_new_inputs` is a right method.

File: ngraph/test/op.cpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/test/op.cpp:134]: At the current moment I removed all places with `copy_with_new_args` these commented code was a PoC to mock node object and this method. The new method `clone_with_new_inputs` is covered by tests.

**Pull request merged**